miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java

```diff
@@ -222,11 +216,21 @@ public Long create(Sample sample) throws IOException {
     if (isDetailedSample(sample)) {
       if (!isIdentitySample(sample)) {
         DetailedSample detailed = (DetailedSample) sample;
-        try {
-          detailed.setParent(findOrCreateParent(detailed));
-          detailed.inheritPermissions(detailed.getParent());
-        } catch (MisoNamingException e) {
-          throw new IOException(e.getMessage(), e);
+        if (detailed.getParent() == null) {
+          throw new IllegalArgumentException("Detailed sample is missing parent identifier");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Detailed sample is missing parent identifier" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)